mzLib/Omics/BioPolymer/SequenceVariation.cs

```diff
@@ -1,167 +1,857 @@
+        /// Order-insensitive multiset comparison of modification dictionaries.
         /// </summary>
-        /// <returns></returns>
+        private static bool ModificationDictionariesEqual(
+            Dictionary<int, List<Modification>> a,
+            Dictionary<int, List<Modification>> b)
+        {
+            if (ReferenceEquals(a, b))
+                return true;
+            if (a is null || b is null)
+                return false;
+            if (a.Count != b.Count)
+                return false;
+
+            // Compare position sets
+            if (!a.Keys.OrderBy(k => k).SequenceEqual(b.Keys.OrderBy(k => k)))
+                return false;
+
+            foreach (var pos in a.Keys)
+            {
+                var listA = a[pos];
+                var listB = b[pos];
+
+                if (listA is null && listB is null)
+                    continue;
+                if (listA is null || listB is null)
+                    return false;
+                if (listA.Count != listB.Count)
+                    return false;
+
+                // Build frequency maps for multiset compare
+                var freqA = listA
+                    .GroupBy(m => m.IdWithMotif ?? m.OriginalId ?? m.ToString())
+                    .ToDictionary(g => g.Key, g => g.Count(), StringComparer.Ordinal);
+                var freqB = listB
+                    .GroupBy(m => m.IdWithMotif ?? m.OriginalId ?? m.ToString())
+                    .ToDictionary(g => g.Key, g => g.Count(), StringComparer.Ordinal);
+
+                if (freqA.Count != freqB.Count)
+                    return false;
+
+                foreach (var kv in freqA)
+                {
+                    if (!freqB.TryGetValue(kv.Key, out int countB) || countB != kv.Value)
+                        return false;
+                }
+            }
+
+            return true;
+        }
+
+        #endregion
+
+        #region Convenience / Interval Logic
+
+        /// <summary>Simple concatenated representation (Original + Begin(+/-End) + Variant).</summary>
         public string SimpleString()
         {
-            return OriginalSequence + OneBasedBeginPosition.ToString() + VariantSequence;
+            if (OneBasedBeginPosition == OneBasedEndPosition || (OriginalSequence?.Length ?? 0) <= 1)
+            {
+                return $"{(OriginalSequence ?? string.Empty)}{OneBasedBeginPosition}{(VariantSequence ?? string.Empty)}";
+            }
+            return $"{(OriginalSequence ?? string.Empty)}{OneBasedBeginPosition}-{OneBasedEndPosition}{(VariantSequence ?? string.Empty)}";
         }
 
+        internal bool Intersects(SequenceVariation segment) =>
+            segment.OneBasedEndPosition >= OneBasedBeginPosition && segment.OneBasedBeginPosition <= OneBasedEndPosition;
+
+        internal bool Intersects(TruncationProduct segment) =>
+            segment.OneBasedEndPosition >= OneBasedBeginPosition && segment.OneBasedBeginPosition <= OneBasedEndPosition;
```

Comment:
These Intersects methods will only return true if the segment is a subset of this SequenceVariation, but not if they merely overlap, e.g.,
`segment.OneBasedEndPosition >= OneBasedBeginPosition && segment.OneBasedEndPosition  <= OneBasedEndPosition && segment.OneBasedBeginPosition  OneBasedEndPosition`

Is this the intended behavior?